src/db/mod.rs

```diff
@@ -0,0 +1,137 @@
+use crate::common::DatabaseError;
+use crate::{audit::log_sync, types::Serializable};
+use sled::Tree;
+
+/// Retrieve a value by key from a tree.
+async fn get_from_tree(db: &Tree, key: &str) -> Result<Vec<u8>, DatabaseError> {
+    match db.get(key) {
+        Ok(result) => match result {
+            Some(value) => Ok(value.to_vec()),
+            None => Err(DatabaseError::NotFound),
+        },
+        Err(_error) => Err(DatabaseError::Get),
+    }
+}
```

Comment:
check https://docs.rs/thiserror/latest/thiserror/
crate, it can help with error conversion.

then it can be refactored to something like
```rust
async fn get_from_tree(db: &Tree, key: &str) -> Result<Vec<u8>, DatabaseError> {
    db.get(key)?.ok_or(DatabaseError::NotFound)
}
```

Reply:
I am already using thiserror.

Reply:
e889da77a756abf1d3390a98806640e31f58cfb7